docs/design/build-system.md

```diff
@@ -0,0 +1,154 @@
+
+In broad strokes the main areas are:
+- loading and processing sources
+- code splitting
+- production optimizations
+- post-processing steps
+
+Many tools are used in this process, and most of these steps happen within a [Webpack](https://webpack.js.org/) build.
+We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
+
+
+## Loading and processing sources
+
+Sources for Angular CLI browser apps are comprised of TypeScript files, style sheets, assets, scripts, and third party dependencies.
+A given build will load these sources from disk, process them, and bundle them together.
+
+
+### TypeScript
+
+Angular builds rely heavily on TypeScript-specific functionality for Ahead-of-Time template compilation (AOT).
+Outside Angular CLI, this is performed by the Angular Compiler (`ngc`), provided by `@angular/compiler-cli`.
+To avail of Ahead-of-Time template compilation within a Webpack compilation we use and distribute the `@ngtools/webpack` Webpack plugin.
+
+Typescript sources are loaded from disk and compiled in-memory into Javascript files that are stored in a virtual file system and made available to Webpack.
+During compilation we also perform a number of code transformations using TypeScript transformers that enable automatic usage of AOT, internationalization features, and server-side rendering.
+
+AOT compilation requires loading HTML and CSS resources referenced on Angular Components. These are processed using a Webpack child compilation.
+
+The build system allows specifying replacements for specific files by replacing what path is loaded from the virtual file system.
+This is used for conditional loading of code at build time.
+
+### Stylesheets
+
+Two types of stylesheets are used in the build system: global stylesheets and component stylesheets.
+Global stylesheets are injected into the `index.html` file, while component stylesheets are loaded directly into compiled Angular components.
+
+The build system supports plain CSS stylesheets as well as the Sass, LESS and Stylus CSS pre-processors.
+Stylesheet processing functionality is provided by `sass-loader`, `less-loader`, `stylus-loader`, `postcss-loader`, `postcss-import`, augmented in the build system by custom webpack plugins.
+
+todo: style resources
+
+
+### Assets
+
+Assets in the build system refer specifically to a list of files or directories that are meant to be copied verbatim as build artifacts.
+These files are not processed and commonly include images, favicons, pdfs and other generic file types.
+They are loaded into the compilation using `copy-webpack-plugin`.
+
+
+### Scripts
+
+Scripts in the build system refer specifically to Javascript files that are meant to be loaded directly on `index.html` without being processed.
+They are loaded into the compilation using a custom webpack plugin.
+
+
+### Third party dependencies
```

Comment:
Are there any differences between Angular dependencies and standard NPM dependencies? I understand that Angular has a special format for distribution/consumption, so I'm curious if there is a separate pipeline there.

Reply:
No real differences. Angular deps are encouraged to follow APF (see below) but APF is geared for maximum compatibility anyway.

Added a mention to this section:

> This resolution strategy supports the [Angular Package Format](https://docs.google.com/document/d/1CZC2rcpxffTDfRDs6p1cfbmKNLA6x5O-NtkJglDaBVs/edit#heading=h.k0mh3o8u5hx).

Reply:
Gotcha, thanks for the reference. Good to call that out here.